apps/progs.pl

```diff
@@ -95,19 +95,31 @@
         genrsa   => "rsa",
         rsautl   => "rsa",
         gendsa   => "dsa",
-        dsaparam => "dsa",
         gendh    => "dh",
-        dhparam  => "dh",
         ecparam  => "ec",
         pkcs12   => "des",
     );
+    my %cmd_deprecated = (
+        dsaparam => [ "3_0", "pkeyparam", "dsa" ],
+        dhparam  => [ "3_0", "pkeyparam", "dh" ],
+    );
```

Comment:
I would actually also deprecate "dsa" and "dh" in favor of "pkey", and also "gendsa" in favor of "genpkey"

Reply:
"dsa", "dh" and "gendsa" all work as is.  Only "gendsa" required changes to work with the deprecations.

I've no objection to deprecating them too, just noting that they weren't a problem converting.

Reply:
I'm willing to add the warnings here but the deprecation of the command line apps and the consequences should be dealt with in a separate PR and scheduled via the tracking spreadsheet.  I'm expecting this will turn into a can of worms pretty quickly.

Reply:
I've noted the app deprectation as a TODO comment and added a task to the sheet.